packages/protocol/contracts/identity/Attestations.sol

```diff
@@ -209,13 +230,24 @@ contract Attestations is
       );
     }
 
+    if(state.requests[msg.sender].blockNumber != 0) {
+      require(
+        !isAttestationTimeValid(accounts[msg.sender].mostRecentAttestationRequest),
+        "Currently active attestation request that has to be revealed first"
+      );
+    }
+
+
+    state.requests[msg.sender].blockNumber = uint128(block.number);
+    state.requests[msg.sender].attestationsRequested = uint128(attestationsRequested);
+
     // solhint-disable-next-line not-rely-on-time
     accounts[msg.sender].mostRecentAttestationRequest = uint96(now);
     accounts[msg.sender].attestationRequestFeeToken = attestationRequestFeeToken;
 
-    IdentifierState storage state = identifiers[identifier];
-
-    addIncompleteAttestations(attestationsRequested, state.attestations[msg.sender]);
+    state.attestations[msg.sender].requested = uint32(
```

Comment:
Yeah I think this needs to happen here as we don't want people to be able to request attestations, but then not select their issuers because they don't like the selection.